src/mlpack/core/optimizers/cne/cne_impl.hpp

```diff
@@ -0,0 +1,205 @@
+    arma::mat& answer)
+{
+  // Get the number of functions to iterate
+  const size_t numFun = function.NumFunctions();
+
+  // Set the population size and fill random values [0,1]
+  population.set_size(populationSize, answer.n_rows);
+  population.randu();
+
+  // initialize helper variables
+  arma::vec parameters(answer.n_rows);
+  fitnessValues.set_size(populationSize);
+  double fitness = 0;
+  double lastBestFitness;
+
+  Log::Info << "CNE initialized successfully. Optimization started"
+  << std::endl;
+
+  // Iterate till max number of generations
+  for (size_t gen = 1; gen <= maxGenerations; gen++)
+  {
+    // calculate fitness values of all candidates
+    for (size_t i = 0; i < populationSize; i++)
+     {
+       // select a candidate
+       parameters = population.row(i).t();
+       // Insert the parameters in the function
+       answer = parameters;
+
+       // find the fitness
+       for (int j = 0; j < numFun; j++)
+       fitness += function.Evaluate(parameters, j);
+
+       // Save fitness values
+       fitnessValues[i] = fitness;
+       fitness = 0;
+     }
+
+      Log::Info << "Generation number: " << gen << " best fitness = "
+      << (double)fitnessValues.min() << std::endl;
+
+      // create the next generation of species
+      Reproduce();
+
+      // check for termination
+      if (finalValue >= (double)fitnessValues.min())
+        {
+          Log::Info << "Terminating. Given fitness criteria " << finalValue
+          << " > " << (double)fitnessValues.min() << std::endl;
+          break;
+        }
+
+      // check for termination
+      if (fitnessHist != DBL_MAX && gen != 1 &&
+        (lastBestFitness - (double)fitnessValues.min()) < fitnessHist)
+        {
+          Log::Info << "Terminating. Fitness History change "
+          << (lastBestFitness - (double)fitnessValues.min())
+          << " < " << fitnessHist << std::endl;
+          break;
+        }
+
+      // Store the best fitness of this generation before update
+      lastBestFitness = (double)fitnessValues.min();
+  }
+
+  // Set the best candidate into the network
+  answer = population.submat(index[0], 0, index[0], populationSize-1);
+
+  // find the best fitness
+  for (int j = 0; j < numFun; j++)
+      fitness += function.Evaluate(answer, j);
+
+  return fitness;
+}
+
+//! Reproduce candidates to create the next generation
+void CNE::Reproduce()
+{
+  // Sort fitness value. The smaller the better
+  index = arma::sort_index(fitnessValues);
+
+  // Find the number of elite percentage
+  size_t numElite = floor(selectPercent * populationSize);
+
+  // Making sure we have even number of candidates to remove and create
+  if ((populationSize - numElite) % 2 != 0) numElite++;
+
+  for (size_t i = numElite; i < populationSize-1; i++)
+  {
+    // Select 2 parents from the elite group randomly [0, numElite)
+    size_t mom = mlpack::math::RandInt(0, numElite);
+    size_t dad = mlpack::math::RandInt(0, numElite);
```

Comment:
@zoq 
u mean the same parent last time? as if select population will be just 10% this will cause a problem .. and same parent is bound to happen. Did you mean same pair of parents ?
sorry im not clear with your words..